index.html

```diff
@@ -1690,59 +1690,93 @@ <h3>Generate Proof</h3>
-not the digital proof on <var>signed document</var> was verified. Whenever
-this algorithm encodes strings, it MUST use UTF-8 encoding.
+integrity of a <a>secured data document</a> by verifying its
+digital proof. Required inputs are a
+<a>secured data document</a> (<var>securedDocument</var>) and
+<a>proof options</a> (<var>options</var>). A <dfn>verification result</dfn> is
+produced as output.
         </p>
 
-<div class="issue">Specify how the public key can be obtained, through
-some out-of-band process and passed in or it can be retrieved by
-derefencing its URL identifier, etc.</div>
-
         <ol class="algorithm">
           <li>
-Get the <a>public key</a> by dereferencing its URL identifier in the
-<code>proof</code> node of the default graph of <var>signed document</var>.
-Confirm that the <a>unsigned data document</a> that describes the <a>public
-key</a> specifies its controller and that its controllers's URL identifier can
-be dereferenced to reveal a bi-directional link back to the key. Ensure that the
-key's controller is a trusted entity before proceeding to the next step.
+Let <var>proof</var> be set to <var>securedDocument</var>.<var>proof</var>.
+          </li>
+          <li>
+If the <var>proof</var>.<var>type</var>,
+<var>proof</var>.<var>verificationMethod</var>, or
+<var>proof</var>.<var>proofPurpose</var> values are not set, a
+`MALFORMED_PROOF_ERROR` MUST be raised.
+          </li>
+          <li>
+If the cryptographic suite does not support unlinkability and the
+<var>proof</var>.<var>created</var> value is not set, a `MALFORMED_PROOF_ERROR`
+MUST be raised.
+          </li>
+          <li>
+Let <var>unsecuredDocument</var> be a copy of <var>securedDocument</var> with
+the `proof` value removed.
+          </li>
+          <li>
+Let <var>transformedDocument</var> be the result of
+<a href="#dfn-transformation">transforming</a> <var>unsecuredDocument</var>
+according to a <a>transformation algorithm</a> associated with the
+<var>cryptosuite</var> specified in <var>proof</var> and the <var>options</var>
+parameters provided as inputs to the algorithm.
+        </li>
+          <li>
+Let <var>hashData</var> be the result of
+<a href="#dfn-hashing">hashing</a> <var>transformedDocument</var>
+according to a <a>hashing algorithm</a> associated with the
+<var>cryptosuite</var> specified in <var>proof</var> and the <var>options</var>
+parameters provided as inputs to the algorithm.
           </li>
           <li>
-Let <var>document</var> be a copy of <var>signed document</var>.
+Let <var>isProofValid</var> be the result of running the
+<a>proof verification algorithm</a> associated with the <var>cryptosuite</var>
+with the <var>hashData</var> and <var>options</var> parameters provided
+as inputs to the algorithm.
+          </li>
+          <li>
+If the <var>proof</var>.<var>proofPurpose</var> value does not match
+<var>options</var>.<var>proofPurpose</var>, a `MISMATCHED_PROOF_PURPOSE_ERROR`
+MUST be raised.
```

Comment:
This step should be run before any effort is made to verify the proof, i.e., the expected proof purpose should be checked first to find a matching proof. We might want the options name to be "expectedProofPurpose" as well, for clarity.

Reply:
Ok, will look into rewriting the algorithm in this way.

Reply:
Fixed in f3a3d2c92ce1688626b2a423bb6a83da0d0578e4.